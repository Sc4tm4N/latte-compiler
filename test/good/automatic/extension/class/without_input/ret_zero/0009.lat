class Node {
  int a;

  void printMe() {
    printInt(a);
  }

  void selfPrintMe() {
    printInt(self.a);
  }
}

int c(Node node) {
  int previous = node.a;
  node.a = 150;
  return previous;
}

int main() {
  Node node = new Node;
  node.printMe();
  node.a = 10;
  node.printMe();
  node.a++;
  node.printMe();
  printInt(c(node));
  node.printMe();

  Node nnode = new Node;
  nnode.selfPrintMe();
  nnode.a = 10;
  nnode.selfPrintMe();
  nnode.a++;
  nnode.selfPrintMe();
  printInt(c(nnode));
  nnode.selfPrintMe();

  return 0;
}
