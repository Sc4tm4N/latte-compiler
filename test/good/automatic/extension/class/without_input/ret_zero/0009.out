0
10
11
11
150
0
10
11
11
150
